consul/src/test/java/com/linecorp/armeria/internal/consul/EnabledIfDockerIsAvailable.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.jupiter.api.condition.EnabledIf;
+
+@Inherited
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+@EnabledIf(
+        value = "com.linecorp.armeria.internal.consul.ConsulTestBase#isDockerAvailable",
+        disabledReason = "Docker is not available."
+)
+public @interface EnabledIfDockerIsAvailable {
```

Comment:
Just to make sure, is there a way to check if our CI has docker available?
If not, I guess we can just try merging and check gradle enterprise if the test is run

Reply:
Maybe not for self-hosted runners. Let me install it soon.

Reply:
I have just installed docker to our self-hosted runners.